securedrop/sdconfig.py

```diff
@@ -0,0 +1,65 @@
+# -*- coding: utf-8 -*-
+
+import config as _config
+
+MYPY = False
+if MYPY:
+    from typing import List  # noqa: F401
+
+
+class SDConfig(object):
+    def __init__(self):
+        # type: () -> None
+        self.ADJECTIVES = _config.ADJECTIVES
+        self.DATABASE_ENGINE = _config.DATABASE_ENGINE
+        self.DEFAULT_LOCALE = _config.DEFAULT_LOCALE
+        self.FlaskConfig = _config.FlaskConfig
+        self.GPG_KEY_DIR = _config.GPG_KEY_DIR
+        self.JOURNALIST_KEY = _config.JOURNALIST_KEY
+        self.JOURNALIST_TEMPLATES_DIR = _config.JOURNALIST_TEMPLATES_DIR
+        self.JournalistInterfaceFlaskConfig = \
+            _config.JournalistInterfaceFlaskConfig
+        self.NOUNS = _config.NOUNS
+        self.SCRYPT_GPG_PEPPER = _config.SCRYPT_GPG_PEPPER
+        self.SCRYPT_ID_PEPPER = _config.SCRYPT_ID_PEPPER
+        self.SCRYPT_PARAMS = _config.SCRYPT_PARAMS
+        self.SECUREDROP_DATA_ROOT = _config.SECUREDROP_DATA_ROOT
+        self.SECUREDROP_ROOT = _config.SECUREDROP_ROOT
+        self.SESSION_EXPIRATION_MINUTES = _config.SESSION_EXPIRATION_MINUTES
+        self.SOURCE_TEMPLATES_DIR = _config.SOURCE_TEMPLATES_DIR
+        self.STORE_DIR = _config.STORE_DIR
+        self.SUPPORTED_LOCALES = []  # type: List[str]
+        self.SourceInterfaceFlaskConfig = _config.SourceInterfaceFlaskConfig
+        self.TEMP_DIR = _config.TEMP_DIR
+        self.WORD_LIST = _config.WORD_LIST
+        self.WORKER_PIDFILE = _config.WORKER_PIDFILE
+        self.env = _config.env
+        self.os = _config.os
+        # Below 4 attributes are only for non-sqlite env
+        self.DATABASE_USERNAME = ""
+        self.DATABASE_PASSWORD = ""
+        self.DATABASE_HOST = ""
+        self.DATABASE_NAME = ""
```

Comment:
it would be better to have a \_\_setattr__ function that getattr(_config, attr) so the class is transparent. And it should **only** allow getting attributes that are upppercase only and underscore. To eliminate any places in the code that try to access something different from config.py

Reply:
This is required because we can not dynamically add any random attribute to a class when we are also doing type checks on the class. This is the reason to create the `SDConfig` class.

Reply:
I feel it is wrong to write code that is not used (because getattr will actually lookup something else) for the sole purpose of indirectly allowing type checking. It is a hack. If there must be a hack to allow mypy to do its job, it needs to be outside of the code base, not inside the codebase.

Reply:
There is no more `getattr` in any piece of code, one of the latest commit actually removed all such dynamic attribution handling. See https://github.com/freedomofpress/securedrop/pull/3033/commits/0f55e8fa50373fe36e599dd199ecef8fbcbf52d1

Reply:
ah, so I'm back to my original suggestion https://github.com/freedomofpress/securedrop/pull/3033#discussion_r169270000